File: yunj/level1/43_하샤드 수.java
class Solution {
    public boolean solution(int x) {
        boolean answer = true;
        
        int origin = x;
        int d = 0;
        
        while(x != 0){
            d += x % 10;
            x /= 10;            
        }
        
        if (origin % d == 0){
            return true;
        } else{
            return false;
        }
    }
}

File: sangjin/level1/43_하샤드_수.cpp
#include <string>
#include <vector>

using namespace std;

bool solution(int x) {
    bool answer = true;
    int sum = 0;
    int num = x;

    while(num != 0){
        sum += num%10;
        num /= 10;
    }

    answer = x%sum == 0;
    
    return answer;
}